flink-connector/src/main/java/org/apache/flink/connector/delta/sink/SchemaConverter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.delta.sink;
+
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+
+import io.delta.standalone.types.*;
+
+/**
+ * This is a utility class to convert from Flink's specific {@link RowType} into
+ * DeltaLake's specific {@link StructType} which is used for schema-matching comparisons
+ * during {@link io.delta.standalone.DeltaLog} commits.
+ */
+public class SchemaConverter {
+
+    /**
+     * Main method for converting from {@link RowType} into {@link StructType}
+     *
+     * @param rowType Flink's logical type of stream's events
+     * @return DeltaLake's specific type of stream's events
+     */
+    public StructType toDeltaFormat(RowType rowType) {
+        StructField[] fields = rowType.getFields()
+                .stream()
+                .map(rowField -> {
+                    DataType rowFieldType = toDeltaDataType(rowField.getType());
+                    return new StructField(
+                            rowField.getName(),
+                            rowFieldType,
+                            rowField.getType().isNullable());
+                })
+                .toArray(StructField[]::new);
+
+        return new StructType(fields);
+    }
+
+    /**
+     * Method containing the actual mapping between Flink's and DeltaLake's types.
+     *
+     * @param flinkType Flink's logical type
+     * @return DeltaLake's data type
+     */
+    public DataType toDeltaDataType(LogicalType flinkType) {
+        switch (flinkType.getTypeRoot()) {
+            case ARRAY:
+                org.apache.flink.table.types.logical.ArrayType arrayType =
+                        (org.apache.flink.table.types.logical.ArrayType) flinkType;
+                LogicalType flinkElementType = arrayType.getElementType();
+                DataType deltaElementType = toDeltaDataType(flinkElementType);
+                return new ArrayType(deltaElementType, flinkElementType.isNullable());
+            case BIGINT:
+                return new LongType();
+            case BINARY:
+            case VARBINARY:
+                return new BinaryType();
```

Comment:
are you sure there is no additional configs need to be injected specifically for VAR?

Reply:
i'm not sure i get you. The constructor for `io.delta.standalone.types.BinaryType` does not allow for any configuration/additional parametrs.